app/main.py

```diff
@@ -1,2 +1,114 @@
+from typing import Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: any) -> None:
+        self.key = key
+        self.value = value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 2 / 3) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.load_factor = load_factor
+        self.table = [None] * capacity
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        index = self.hash(key)
+        node = self.table[index]
+        if node is None:
+            self.table[index] = Node(key, value)
+            self.size += 1
+        else:
+            while node:
+                if node.key == key:
+                    node.value = value
+                    break
+                if node.next is None:
+                    node.next = Node(key, value)
+                    self.size += 1
+                    break
+                node = node.next
+
+        if self.size / self.capacity >= self.load_factor:
+            self.resize()
+
+    def __getitem__(self, key: Hashable) -> str:
+        index = self.hash(key)
+        node = self.table[index]
+        while node:
+            if node.key == key:
+                return node.value
+            node = node.next
+        raise KeyError(f"Key {key} not found")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def clear(self) -> None:
+        self.table = [None] * self.capacity
+        self.size = 0
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        new_table = [None] * self.capacity
+        for node in self.table:
+            while node:
+                index = self.hash(node.key)
+                if new_table[index] is None:
+                    new_table[index] = Node(node.key, node.value)
+                else:
+                    current = new_table[index]
+                    while current.next:
+                        current = current.next
+                    current.next = Node(node.key, node.value)
+                node = node.next
+        self.table = new_table
```

Comment:
u can write old table to a variable and than iterate through it using method setitem to simplify ur code